src/KeyManagement.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {EnumerableSetLib} from "solady/utils/EnumerableSetLib.sol";
+import {Key, KeyLib, KeyType} from "./libraries/KeyLib.sol";
+import {MinimalDelegationStorage, MinimalDelegationStorageLib} from "./libraries/MinimalDelegationStorage.sol";
+import {IKeyManagement} from "./interfaces/IKeyManagement.sol";
+
+/// @dev A base contract for managing keys.
+abstract contract KeyManagement is IKeyManagement {
+    using EnumerableSetLib for EnumerableSetLib.Bytes32Set;
+    using KeyLib for Key;
+
+    /// @dev Must be overridden by the implementation
+    function _authorizeCaller() internal view virtual {}
+
+    /// @inheritdoc IKeyManagement
+    function authorize(Key memory key) external returns (bytes32 keyHash) {
```

Comment:
should we rename? this is an overused term and I think of this action more as add?

Also with the new hooks, I wonder if we should have a function that also can add(key, validationModule)

Reply:
#61